engine_test.go

```diff
@@ -217,31 +217,41 @@ outputs:
 	assert.Equals(t, outputData.(map[any]any), map[any]any{"message": "Hello, Arca Lot!"})
 }
 
-// Test_CacheSubworkflows tests that every sub-workflow filename
-// referenced in the main workflow is incorporated into the
-// workflow's execution.
-func Test_CacheSubworkflows(t *testing.T) {
-	fileCache, err := loadfile.NewFileCacheUsingContext(
-		"fixtures/test-subworkflow",
-		map[string]string{
-			"workflow": "workflow-happy.yaml",
-		})
-	assert.NoError(t, err)
-	assert.NoError(t, fileCache.LoadContext())
+func TestE2EWorkflowDefaultInput(t *testing.T) {
+	content := map[string][]byte{
+		"workflow.yaml": []byte(`version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        name:
+          type:
+            type_id: string
+          default: not
+          required: false
+steps:
+  example:
+    plugin: 
+      src: quay.io/arcalot/arcaflow-plugin-template-python:0.2.1
+      deployment_type: image
+    step: hello-world
+    input:
+      name: !expr $.input.name
+outputs:
+  success:
+    message: !expr $.steps.example.outputs.success.message`),
+	}
+	fileCache := loadfile.NewFileCache("", content)
 	outputID, outputData, outputError, err := createTestEngine(t).RunWorkflow(
 		context.Background(),
 		[]byte(`{}`),
 		fileCache,
-		"workflow",
+		"",
 	)
 	assert.NoError(t, err)
 	assert.Equals(t, outputError, false)
 	assert.Equals(t, outputID, "success")
-	nStepOutput := 3
-	outputDataTyped := outputData.(map[any]any)
-	assert.Equals(t, len(outputDataTyped), nStepOutput)
-	for _, v := range outputDataTyped {
-		vMapStr := v.(map[string]any)
-		assert.MapContainsKey(t, "success", vMapStr)
-	}
+	assert.Equals(t, outputData.(map[any]any), map[any]any{"message": "Hello, not!"})
```

Comment:
This is only half the test, right?  That is, we need to prove that the default value comes through when it is not overridden, and that it does _not_ get used when it **is** overridden.  (I assume that we already have the "other half"...where is it?)